java-connectors/build.gradle

```
@@ -94,6 +94,8 @@ allprojects {
     }
 
     shadowJar {
+        def artifactVersion = (gitTag == null || gitTag.isBlank()) ?
+                project.version : gitTag
```

I think this would do what you want, but I haven't tried it.

```suggestion
        def artifactVersion = gitTag?.trim() ? gitTag : project.version
```